### PR TITLE
Score choices

### PR DESCRIPTION
**Pull request by gimait**
This is a first iteration of the behavior chooser to select which behaviors will be executed during an activity.

Looking at the information stored in the activity configuration files, I am assuming that the behavior choices are presented as a list of options to try sequentially. This list is always the same on `StrictPriority` items, but can change on each iteration when using `Scoring` choice types. Then, this behavior list is used by the main <thread/brain/application> to select an available behavior and execute it. Once a behavior is executed successfully, the cycle starts over (create behavior list, then run through it).

This pull request only includes the logic to generate the prioritized behavior lists.

### CODE REVIEW

File: pycozmo/activity.py [hunk 1]
```diff
@@ -24,6 +27,7 @@ class BehaviorChooser:
         "choice_type",
         "behaviors",
         "iteration",
+        "__dict__",
```
Comment (zayfod) [inline pycozmo/activity.py:30]: Why is this necessary?

File: pycozmo/activity.py [hunk 2]
```diff
@@ -32,7 +36,10 @@ def __init__(self,
 
         self.choice_type = str(choice_type)
         self.behaviors = behaviors
-        self.iteration = 0
+
+        if self.choice_type == 'Scoring':
+            self.init_scores()
+            self.init_repetition_penalty()
```
Comment (zayfod) [inline pycozmo/activity.py:42]: Is repetition penalty specific to the scoring behavior chooser?

File: pycozmo/activity.py [hunk 3]
```diff
@@ -43,20 +50,59 @@ def from_json(cls, data: Dict):
...
+        self.total_score = sum(self.behavior_scores)
+
+    def init_repetition_penalty(self) -> None:
+        self.repetition_penaltys = []
+        for b in self.behaviors:
+            if 'repetitionPenalty' in b['scoring']:
+                self.repetition_penaltys.append(
+                    DecayGraph([Node(x=n['x'], y=n['y']) for n in b['scoring']['repetitionPenalty']['nodes']]))
+            else:
+                self.repetition_penaltys.append(DecayGraph([Node(x=1, y=0)]))
```
Comment (zayfod) [inline pycozmo/activity.py:72]: 👍

**Review by zayfod — APPROVED** [review]

File: pycozmo/activity.py (hunk 1 above)
Comment (gimait) [reply pycozmo/activity.py:30]: I added it to have as optional variables the `behavior_scores`, `behavior_names`, `behavior_repetitions` and `repetition_penaltys`. Done this way, they don't need to be defined in the slots. I thought it made sense since these variables are generated for the scored choosers only, but I can see that there are arguments for specifying them in slots directly or not using slots at all here. What do you prefer? I'm still learning to use `__slots__` vs `__dict__` to define variables in these classes (always used the default, `__dict__` before), so I would appreciate to know what would you do, and what you think makes more sense and/or is faster/more efficient.

File: pycozmo/activity.py (hunk 2 above)
Comment (gimait) [reply pycozmo/activity.py:42]: Yes, it is (as long as I have seen). There is only one json with 'Selection' type, which does not contain any other info (actvitities_config.json), and the 'Strict' types always return the same behavior progression.
